src/client/datascience/jupyter/kernels/providers/installedLocalKernelProvider.ts

```diff
@@ -39,11 +39,8 @@ export class InstalledLocalKernelSelectionListProvider
                         // If it is, then no need to display that (selecting kernels registered is done by selecting the corresponding interpreter).
                         // Hence we can hide such kernels.
                         // Kernels we create will end with a uuid (with - stripped), & will have interpreter info in the metadata.
-                        if (
-                            item.metadata?.interpreter &&
-                            item.name.length > 32 &&
-                            item.name.slice(-32).toLowerCase() === item.name
-                        ) {
+                        const guidRegEx = /[a-f0-9]{32}$/;
+                        if (item.metadata?.interpreter && item.name.toLowerCase().search(guidRegEx)) {
```

Comment:
I'm not fully sure what the old check was trying here, but I don't think that it would have ever worked as a substring slice would never be equal to the full string. My one slight question here is if we still fill out the interpreter metadata in kernelspecs or if that was just in the old code. If it's not just in the old code this does mean that if you create a kernelspec with a name that ends in a guid we would exclude it, but I believe that we only fill out the interpreter metadata in the old code paths.

Reply:
With ZMQ off I found that these kernels are still created. They're created for every interpreter that you try to use. So it would be weird if they didn't show up?

Reply:
The old code is obviously wrong but not sure what the side effects are.

Reply:
@rchiodo Good call, it's making the code work the right way. Lemme verify quick.

Reply:
@rchiodo I was trying to trace all the paths here, then I realized that just doing this might be cleaner. Only hide the old interpreters in the raw scenarios. Keeps the old paths the same for when they register interpreters. Does introduce a new concept into the kernel provider code, but that doesn't seem too bad as that's the whole point of having these small services.

Reply:
@rchiodo were you ok with the change here? Didn't want to commit otherwise.

Reply:
Yeah just wanted to make sure you tested it.

Reply:
Gave it a whirl. Looks like I see the expected specs / interpreters on there with ZMQ off. However I am seeing some issues there with kernel switching. If I repro those same issues without my change here then I'll just check in and file that as a new issue.